Assets/Scripts/PlayerSupervisor.cs

```diff
@@ -23,11 +56,129 @@ void Update()
+            totalAddedWidth = blockWidth + spaceBetweenBlocks*2;
+        } 
+        else 
+        {   
+            blockWidth = randNum;  
+            totalAddedWidth = blockWidth + spaceBetweenBlocks;        
+        }
+        float[] returnArr = new float[2];
+        returnArr[0] = blockWidth;
+        returnArr[1] = totalAddedWidth;
+        return returnArr;
+    }
+
+    // Frannie - helper for dynamic block function
+    private GameObject createBlock(float blockMidPoint, float blockRow)
     {
-        points += pointValue;
-        gameManager.UpdatePoints(points);
+        // create new game object
+        // x = midpoint position of block in screen space
+        // y = current row position
+        GameObject newBlock = Instantiate(block, 
+                    new Vector2(blockMidPoint, blockRow), 
+                    Quaternion.identity, 
+                    parentBlock.transform // need this to it groups under "Blocks"
+        );
+        return newBlock;
+    }
+
+    // Frannie's Level - dynamically and randomly setting up blocks
+    // each block's width (column space) is what will be random - always will be 5 rows
+    // block vector position is the MIDDLE OF THE BLOCK
+    public void CreateStartingBlocks()
+    {
+        float randNum;
+        int blocksPlaced;
+        float blockRow = blockStartRow;
+        float blockMidpoint;
+        float[] widths;
+        float rowWidth;
+        float blockWidth;
+        float priorBlockWidth; 
+        float priorBlockMidpoint; 
+        Vector3 scaleChange;
+        for(int i = 0; i < 5; i ++) // 5 rows
+        {
+            // curr row start position (i.e. y value for block)
+            blockRow += (1f + spaceBetweenBlocks);
 
+            // reset these values each iteration
+            blocksPlaced = 0;  
+            rowWidth = 0;
+            blockMidpoint = 0;
+            priorBlockWidth = 0;
+            priorBlockMidpoint = 0;
+
+            // While there is still space left in current row
+            // ROW WIDTH INCLUDES SPACE BETWEEN BLOCKS AND BLOCK WIDTHS
+            while(rowWidth < screenWidthWorld){
+
+                randNum = UnityEngine.Random.Range(minBlockWidthWorld, maxBlockWidthWorld);
+                widths = calculateWidths(randNum, rowWidth);
+                blockWidth = widths[0]; // block widths in position 0
+                blockMidpoint = calculateBlockMidpoint(blocksPlaced, blockWidth, 
+                                                       priorBlockWidth, priorBlockMidpoint);
+                priorBlockWidth = blockWidth;
+                priorBlockMidpoint = blockMidpoint;
+                GameObject newBlock = createBlock(blockMidpoint, blockRow);
+
+                // update block scale based on block width, note scale is a PERCENTAGE
+                scaleChange = new Vector2(blockWidth/prefabBlockWidthWorld - 1, 0);
+                newBlock.transform.localScale += scaleChange;

```

Comment:
This was a little hard to follow - I think mostly because of the tricky math. It might make more sense here to remove the -1 and make the localScale a straight set rather than +=. I think the math here relies on localScale always starting as 1 so the initial value and the -1 seem to be a wash, but I'm sure you understand what's going on here better than I do and trust your judgement if there's a case for using the initial localScale value when resizing. 
```suggestion
                scaleChange = new Vector2(blockWidth/prefabBlockWidthWorld, 0);
                newBlock.transform.localScale = scaleChange;

```

Reply:
agreed this is confusing, changed to your suggestion!